src/client.rs

```diff
@@ -144,17 +140,17 @@ impl ClientFlow {
 
     async fn progress_send(&mut self) -> Result<Option<ClientFlowEvent>, ClientFlowError> {
         match self.send_command_state.progress(&mut self.stream).await? {
-            Some(SendCommandEvent::CommandSent {
+            Some(SendCommandEvent::Command {
                 key: handle,
                 command,
             }) => Ok(Some(ClientFlowEvent::CommandSent { handle, command })),
-            Some(SendCommandEvent::CommandAuthenticateStarted { key: handle }) => {
+            Some(SendCommandEvent::CommandAuthenticate { key: handle }) => {
```

Comment:
I think we could align the naming of `SendCommandEvent` and `ClientFlowEvent` even more. But maybe we should first get everything in the public API as we want it. Maybe even using more `enum`s like `ClientFlowEvent::{Sent,Received,Authenticate,Idle}` and align the internals afterwards.

Nothing to do here, only a thought.